app.py

```diff
@@ -1036,6 +1011,32 @@ def main_page(page):
                            current_page='start', content=main_page_translations)
 
 
+@app.route('/for-teachers', methods=['GET'])
+@requires_login
+def for_teachers_page(user):
+    if not is_teacher(user):
+        return utils.error_page(error=403, ui_message=gettext('not_teacher'))
+
+    page_translations = hedyweb.PageTranslations('for-teachers').get_page_translations(g.lang)
+    welcome_teacher = session.get('welcome-teacher') or False
+    session.pop('welcome-teacher', None)
+
+    teacher_classes = DATABASE.get_teacher_classes(current_user()['username'], True)
```

Comment:
This is guaranteed by the `@requires_login` function call at the top of this route. This makes sure that we need to be a logged in user to visit this path, otherwise we are automatically re-directed.